### ISSUE TITLE
Get the send-more-money code builder macro to work

### ISSUE THREAD

**Issue by masak** [masak/007#494]
I was toying around with [this code](https://github.com/masak/007/blob/1736eac0a1050a7b5e0ee3623101adffd07b3684/examples/send-more-money/stage-3.007.wip) yesterday, and it felt like I _almost_ got it to work. A few things are amiss:

* For some reason, `Q.Block`s get inserted everywhere. I don't remember the reason for that, but it's annoying.
* I want to insert a `Q.StatementList`. Probably I need to declare that with `{{{Q.StatementList @ result}}}`, otherwise the parser won't know to do the right thing. (In this case it feels there _might_ be leeway, but my brain is not equipped to think about it since it ultimately involves parser states.)
* It'd be nice if it worked to send the `{{{Q.StatementList @ result}}}` an `Array` of `Q.Statement` just like that &mdash; that is, there's convenient auto-wrapping going on for `Q.StatementList` and all the `Q.*List` types from an `Array` to a `Q.*List`.

In short though, make that example work (which would be _awesome_), or find a hard proof of its impossibility.

In general, I felt that the ergonomics of writing this kind of macro is not there at all yet. But that's not so strange. This is the first time I even think to write a macro like this. I'll stick to filing concrete issues, and hopefully it'll just converge on something usable.

**Comment by masak** [comment]
> * For some reason, `Q.Block`s get inserted everywhere. I don't remember the reason for that, but it's annoying.

I was hitting this again as I was starting in on #379.

```
$ bin/007 -e='say(type(quasi { func foo() {} }))'
<type Q.Block>
```

I think I would be much happier if it said `<type Q.Statement.Func>`.
